src/Crypto/WebAuthn/Cose/Internal/Verify.hs

```diff
@@ -87,21 +90,22 @@ fromX509 key = Left $ "X509 public key algorithm is not supported: " <> Text.pac
 -- 'Cose.PublicKeyWithSignAlg' Returns an error if the signature algorithm
 -- doesn't match. Also returns an error if the signature wasn't valid or for
 -- other errors.
-verify :: Cose.PublicKeyWithSignAlg -> BS.ByteString -> BS.ByteString -> Either Text ()
+-- FIXME: https://w3c.github.io/webauthn/#sctn-signature-attestation-types kind of documents this, but not for all formats. This is notably not really related to COSE, but rather webauthn's own definitions. The spec should be made less ambiguous, file upstream issues and refactor this code
+verify :: Cose.PublicKeyWithSignAlg -> CoseMessage -> CoseSignature -> Either Text ()
```

Comment:
@lykahb I think the need for this PR actually highlights a bigger problem, the one indicated in this FIXME here. The spec notably doesn't really define all binary formats it refers to. Notably I discovered that the message and signature types here are actually a custom format, meaning that they should probably be called `WebAuthnMessage`/`WebAuthnSignature` instead, and this whole module should be moved outside of the Cose module tree.

And the specification is really unclear about this in a bunch of places, I'd like to file an issue in the spec for addressing this.

Reply:
With these holes in the spec, we kind of need to reverse-engineer what the spec really means based on implementations.

Reply:
Yeah, that's a large and not well defined scope. Can we split the changes? The `ByteString` -> `Base16ByteString` that kills the orphan instance, a new type for the PNG data URL, and the comments on the `ToJSON` would be great to have sooner.